source/guides/postgres-migration.rst

```diff
@@ -0,0 +1,204 @@
+        preserve index names
+
+   SET PostgreSQL PARAMETERS
+        maintenance_work_mem to '128MB',
+        work_mem to '12MB'
+
+   SET MySQL PARAMETERS
+         net_read_timeout  = '120',
+         net_write_timeout = '120'
+
+   CAST column Channels.Type to channel_type drop typemod,
+        column Teams.Type to team_type drop typemod,
+        column UploadSessions.Type to upload_session_type drop typemod,
+        column Drafts.Priority to text,
+        type int when (= precision 11) to integer drop typemod,
+        type bigint when (= precision 20) to bigint drop typemod,
+        type text to varchar drop typemod,
+        type tinyint when (<= precision 4) to boolean using tinyint-to-boolean,
+        type json to jsonb drop typemod
+
+   MATERIALIZE VIEWS exclude_products
+        excluding table names matching ~<IR_>, ~<focalboard>
+
+   BEFORE LOAD DO
+        $$ ALTER SCHEMA public RENAME TO {{ .source_schema }}; $$
+
+   AFTER LOAD DO
+        $$ UPDATE {{ .source_schema }}.db_migrations set name='add_createat_to_teamembers' where version=92; $$,
+        $$ ALTER SCHEMA {{ .source_schema }} RENAME TO public; $$;
+
+Once you save this configuration file eg. ``migration.load``, you can run the ``pgLoader`` with following command:
+
+.. code:: bash
+
+   pgLoader migration.load > migration.log
+
+Feel free to contribute and/or report your findings through the migration.
+
+Compare the data
+----------------
+
+We internally developed a tool to simplify the process of comparing contents of two databases. The ``dbcmp`` tool compares every table and reports whether if there is a diversity between two schemas.
+
+The tool has a few flags needs to be supplied to run a comparison:
+
+.. code:: sh
+
+   Usage:
+     dbcmp [flags]
+
+   Flags:
+         --exclude strings   exclude tables from comparison, takes comma-separated values.
+     -h, --help              help for dbcmp
+         --source string     source database dsn
+         --target string     target database dsn
+     -v, --version           version for dbcmp
+
+For our case we can simply run the following command:
+
+.. code:: sh
+
+   dbcmp --source "${MYSQL_DSN}" --target "${POSTGRES_DSN}" --exclude="db_migrations","ir_","focalboard","systems"
+
+Note that the migration guide only covers the tables for Mattermost channels, the support for other plugins such as Boards and Playbooks will be added in the future. Another exlusion we are making is in the ``db_migrations`` table which has a small difference (a typo in a single migration name) creates a diff. Since we created the Postgres schema with morph and the official mattermost source, we can consider to skip it safely. On the other hand, ``systems`` table may contain additional diffs if there was extra keys added during some of the migrations. Consider excluding ``systems`` table if you run into issues and do a manual comparison as the data in the ``systems`` table is relatively smaller in size.
```

Comment:
```suggestion
Note that this migration guide only covers the tables for Mattermost channels. Support for other plugins, such as Playbooks, will be added in the future. 

Another exclusion we are making is in the ``db_migrations`` table which has a small difference (a typo in a single migration name) creates a diff. Since we created the PostgreSQL schema with morph, and the official ``mattermost`` source, we can skip it safely without concerns. On the other hand, ``systems`` table may contain additional diffs if there were extra keys added during some of the migrations. Consider excluding the ``systems`` table if you run into issues, and perform a manual comparison as the data in the ``systems`` table is relatively smaller in size.
```